### PR TITLE
Implement ESLinter

### PR DESCRIPTION
**Pull request by carlxw**
Hi everyone, prior to merging, please check if these rules are what we want to be enforced. Everybody does have different coding practices, so please take the time to review what we should add/remove before merging this into main.

Here's some good references:
* https://discordjs.guide/preparations/setting-up-a-linter.html#setting-up-eslint-rules
* https://eslint.org/docs/latest/rules/

### CODE REVIEW

**Comment by sherwinchiu** [comment]
Task linked: [CU-8687acxtz linting for JS consistency across the repo](https://app.clickup.com/t/8687acxtz)

**Comment by sherwinchiu** [comment]
can we just do the default for ESLinter? i think that should be good enough, and then we can go from there

**Comment by carlxw** [comment]
@sherwinchiu What exactly do you mean by "default"?

**Comment by carlxw** [comment]
TLDR
* Added ESLinter with JS Code Standard with a few other rules
* Added Husky to lint pre-commit
  * This feature is currently disabled because ALL files are going to need lots of work before re-enabling this
